server/trpc/router/sendEmail.ts

```diff
@@ -0,0 +1,19 @@
+import toast from "react-hot-toast";
+import { SendEmailSchema } from "../../../schema/comment";
+import sendEmail from "../../../utils/sendEmail";
+import { protectedProcedure, router } from "../trpc";
+
+export const sendEmailRouter = router({
+
+    send:protectedProcedure
+        .input(SendEmailSchema)
+        .mutation(async ({input}) => {
+            const { recipient, htmlMessage, subject } = input;
+            sendEmail({
```

Comment:
```
try {
  await sendEmail({
      recipient: process.env.ADMIN_EMAIL || '',
      htmlMessage,
      subject,
  });
} catch (error) {
  throw new Error(`Failed to send email: ${error.message}`);
}
```

Reply:
try this to see if it trigger onError

Reply:
I tried this exact thing, but it did not work. I also tried the same on the sendEmail function in /utils/sendEmail, but that did not work either. I will experiment more with wrapping a promise.
Thanks for all the feedback. Its very helpful.